syntax = "proto3";

package auth;

service AuthService {
    rpc Register(RegisterRequest) returns (AuthResponse);
    rpc Login(LoginRequest) returns (AuthResponse);
    rpc Logout(LogoutRequest) returns (AuthResponse);
}

message RegisterRequest {
    string username = 1;
    string password = 2;
    string email = 3;
}

message LoginRequest {
    string username = 1;
    string password = 2;
}

message LogoutRequest {
    string username = 1;
}

message AuthResponse {
    bool success = 1;
    string message = 2;
    string token = 3;
}
